core/src/test/java/BookServiceTest.java

```diff
@@ -0,0 +1,200 @@
+
+        bookService.cancelRequestsByIdBook(book.getId());
+        assertThat(book.getStatus()).isEqualTo(BookStatus.IN_STOCK);
+        verify(bookRepository, times(requestList.size())).findById(notNull());
+        verify(requestRepository, times(1)).findAll();
+        verify(requestRepository, times(1)).delete(notNull());
+    }
+
+    @Test
+    void getDescriptionOfBook() {
+        Book book = bookList.get(0);
+        when(bookRepository.findById(anyInt())).thenReturn(Optional.of(book));
+        String description = bookService.getDescriptionOfBook(anyInt());
+        Assertions.assertEquals(description, "The best book 2022 year");
+        verify(bookRepository, times(1)).findById(notNull());
+    }
+
+    @Test
+    void getStaleBooks() {
+        List<Order> completedOrderList = orderList.stream()
+                .filter(order -> order.getStatus().equals(OrderStatus.COMPLETED))
+                .toList();
+        List<Book> staleBooks;
+        when(bookRepository.findAll()).thenReturn(bookList);
+        when(orderRepository.getCompletedOrdersForPeriod(any(), any())).thenReturn(completedOrderList);
+
+        staleBooks = bookService.getStaleBooks();
+        MatcherAssert.assertThat(staleBooks, hasSize(5));
+        verify(bookRepository, times(1)).findAll();
+        verify(orderRepository, times(1)).getCompletedOrdersForPeriod(notNull(), notNull());
+    }
+
+    @Test
+    void getBooksSortedByTitleAlphabetically() {
+        when(bookRepository.findAll()).thenReturn(bookList);
+        List<Book> sortedListBooks = bookService.getBooksSortedByTitleAlphabetically();
+        Assertions.assertEquals(sortedListBooks.get(2).getId(), 4);
+        verify(bookRepository, times(1)).findAll();
+    }
+
+    @Test
+    void getBooksSortedByDatePublication() {
+        when(bookRepository.findAll()).thenReturn(bookList);
+        List<Book> sortedListBooks = bookService.getBooksSortedByDatePublication();
+        Assertions.assertEquals(sortedListBooks.get(1).getId(), 6);
+        verify(bookRepository, times(1)).findAll();
+    }
+
+    @Test
+    void getBooksSortedByPrice() {
+        when(bookRepository.findAll()).thenReturn(bookList);
+        List<Book> sortedListBooks = bookService.getBooksSortedByPrice();
+        Assertions.assertEquals(sortedListBooks.get(1).getId(), 2);
+        verify(bookRepository, times(1)).findAll();
+    }
+
+    @Test
+    void getBooksSortedByAuthor() {
+        when(bookRepository.findAll()).thenReturn(bookList);
+        List<Book> sortedListBooks = bookService.getBooksSortedByAuthor();
+        Assertions.assertEquals(sortedListBooks.get(1).getId(), 5);
+        verify(bookRepository, times(1)).findAll();
+    }
+
+
+    private void initService() throws NoSuchFieldException, IllegalAccessException {
+        Class bookServiceClass = this.bookService.getClass();
+
+        Field isAutoRequestClosed = bookServiceClass.getDeclaredField("isAutoRequestClosed");
```

Comment:
Using ReflectionUtils or reflection in tests is generally discouraged due to several reasons. Reflection in Java provides dynamic access to fields, methods, and constructors during runtime, but it comes with trade-offs that can impact code readability, maintainability, and robustness. Here are some key reasons why it's not recommended:
   - Code that relies heavily on reflection can be difficult to read and understand, making it harder for developers to grasp the logic and intentions behind the tests.
   - Reflection bypasses the usual compile-time checks, making the code more prone to errors. If the structure of the classes being reflected upon changes, it can break the tests, and these issues might not be caught until runtime.
   - Many testing tools and frameworks are designed to work seamlessly with standard testing practices. Reflection-based code may not be as compatible with these tools, leading to a lack of support for automated testing and other testing utilities.
   - Debugging reflective code can be more challenging than debugging regular code. Developers may find it harder to trace and understand the flow of the program during debugging sessions.
   - Reflection can have performance overhead, and using it excessively in tests may impact the overall test suite performance. Tests should ideally run quickly and efficiently.
   - The need for extensive use of reflection in tests might indicate design issues in the codebase. It's often better to address these design concerns rather than resorting to reflection for testing purposes.

Reply:
To fix issue with value you can use test resources, with desired data

@TestPropertySource(locations = "classpath:test-application.properties")

Reply:
I fixed, but I am not sure it is easy for understanding